### PR TITLE
Http connector follow redirect for maximum.redirects, also http acceptor send 302 for wsn and wsx

### PR DESCRIPTION
**Pull request by dpwspoon**
Http connector follow redirect for maximum.redirects, also http acceptor send 302 for wsn and wsx

### CODE REVIEW

File: service/http.balancer/src/test/scripts/org/kaazing/gateway/service/http/balancer/wsn.balancer.request.rpt [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+write version "HTTP/1.1"
+write header host
+write header "Upgrade" "websocket"
+write header "Connection" "Upgrade"
+write header "Sec-WebSocket-Key" "dGhlIHNhbXBsZSBub25jZQ=="
+write header "Sec-WebSocket-Version" "13"
+
+read status "302" /.+/
+read version "HTTP/1.1"
+read header "location" "ws://localhost:8001/echo1"
```
Comment (jitsni) [inline service/http.balancer/src/test/scripts/org/kaazing/gateway/service/http/balancer/wsn.balancer.request.rpt]: should the redirect be of ws scheme url ? shouldn't it be http url ?

File: util/src/main/java/org/kaazing/gateway/util/feature/EarlyAccessFeatures.java [hunk 2]
```diff
@@ -21,5 +21,7 @@
 public interface EarlyAccessFeatures {
 
     static EarlyAccessFeature HTTP_PROXY_SERVICE = new EarlyAccessFeature("http.proxy", "HTTP Proxy Service", false);
+    static EarlyAccessFeature WSN_302_REDIRECT = new EarlyAccessFeature("wsn.302.redirect", "Send redirect for wsn via 302", false);
```
Comment (jitsni) [inline util/src/main/java/org/kaazing/gateway/util/feature/EarlyAccessFeatures.java]: static keyword is unnecessary

File: transport/http/src/main/java/org/kaazing/gateway/transport/http/DefaultHttpSession.java [hunk 3]
```diff
@@ -615,5 +626,22 @@ public boolean isHttpxeSpecCompliant() {
         return httpxeSpecCompliant;
     }
 
+        void resetParent(IoSessionEx newParent){
+        setParent(newParent);
+    }
+
+    int getAndDecrementRedirectsAllowed() {
```
Comment (jitsni) [inline transport/http/src/main/java/org/kaazing/gateway/transport/http/DefaultHttpSession.java]: Is this code needed ?

File: transport/http/src/main/java/org/kaazing/gateway/transport/http/HttpConnector.java [hunk 4]
```diff
@@ -151,24 +173,20 @@ public void run() {
...
     }
 
     protected <T extends ConnectFuture> void connectUsingExistingTransport(final ConnectFuture connectFuture,
                 ResourceAddress address, IoSession transportSession, IoHandler handler,
-                IoSessionInitializer<T> initializer) {
-
-        IoSessionInitializer<T> httpSessionInitializer = createHttpSessionInitializer(handler, initializer);
-
-        Callable<DefaultHttpSession> httpSessionFactory = createHttpSession(address, transportSession, httpSessionInitializer, connectFuture);
+                HttpSessionFactory httpSessionFactory) {
```
Comment (jitsni) [inline transport/http/src/main/java/org/kaazing/gateway/transport/http/HttpConnector.java]: handler parameter is not used in some methods.

File: service/http.balancer/src/test/scripts/org/kaazing/gateway/service/http/balancer/wsn.balancer.request.rpt (hunk 1 above)
Comment (dpwspoon) [inline service/http.balancer/src/test/scripts/org/kaazing/gateway/service/http/balancer/wsn.balancer.request.rpt]: yup, fixing.  Thanks

File: transport/http/src/main/java/org/kaazing/gateway/transport/http/DefaultHttpSession.java (hunk 3 above)
Comment (dpwspoon) [inline transport/http/src/main/java/org/kaazing/gateway/transport/http/DefaultHttpSession.java]: Yeah, without it the parent is wrong and write events are going down the wrong downstream

**Comment by dpwspoon** [comment]
@jitsni This is ready for another round of review

**Pull request merged**